src/kv/untyped_map.h

```diff
@@ -256,6 +260,12 @@ namespace kv::untyped
               changes = true;
               state = state.remove(it->first);
             }
+            else if (track_deletes_on_missing_keys)
```

Comment:
This is a surprising choice, is it necessary for etcd behaviour?

Reply:
This is needed as when we rebuild the map for just the diff we won't have any keys and so need to track the deletes to retain their writes in the commit.

Reply:
Ah ok.